lib/matplotlib/units.py

```diff
@@ -43,12 +43,83 @@ def default_units(x, axis):
 """
 
 from numbers import Number
+import inspect
 
 import numpy as np
 
 from matplotlib.cbook import iterable, safe_first_element
 
 
+def _accepts_units(convert_x=[], convert_y=[]):
+    """
+    A decorator for functions and methods that accept units. The parameters
+    indicated in *convert_x* and *convert_y* are used to update the axis
+    unit information, are converted, and then handed on to the decorated
+    function.
+
+    Parameters
+    ----------
+    convert_x, convert_y : list
+        A list of integers or strings, indicating the arguments to be converted
+    """
+    def decorator(func):
```

Comment:
The current implementation hides the docstring of the decorated functions. Not an expert on decorators, but it might be sufficient to decorate `decorator(func)` with `@wraps`.

Reply:
Doing the `@wraps` seems to work